executor/adapter.go

```diff
@@ -1923,7 +1937,11 @@ func (a *ExecStmt) SummaryStmt(succ bool) {
 	if a.retryCount > 0 {
 		stmtExecInfo.ExecRetryTime = costTime - sessVars.DurationParse - sessVars.DurationCompile - time.Since(a.retryStartTime)
 	}
-	stmtsummary.StmtSummaryByDigestMap.AddStatement(stmtExecInfo)
+	if sessVars.EnablePersistentStmtSummary {
+		sessVars.StmtSummary.Add(stmtExecInfo)
```

Comment:
The code appended around L1811 shows nil checks but not here, why?

Reply:
L1814-L1817 indicate that if `enabled == true`, `sessVars.StmtSummary` must not be nil. Because L1823 will return if `enabled == false`.